packages/navie/src/commands/explain-command.ts

```diff
@@ -188,3 +189,15 @@ export default class ExplainCommand implements Command {
     }
   }
 }
+
+function gathererEnabled(
+  userOptions: UserOptions,
+  agentMode: AgentMode,
+  contextLabels: ContextV2.ContextLabel[]
+): boolean {
+  const enabledByDefault =
+    [AgentMode.Generate, AgentMode.Test].includes(agentMode) ||
+    !!contextLabels.find((l) => l.name === ContextV2.ContextLabelName.Overview);
```

Comment:
I think ultimately, we want the "generate code" behavior where we emit patch files to work in @explain mode when the user's question is classified as `Generate`. Basically, the less we rely on explicit command modes, the better. So if and when that happens, we would also want to "gather" in implicit-generate mode the way that we do in explicit-generate mode.